### ISSUE TITLE
Deserialize private collection/dictionary causes MethodAccessException

### ISSUE THREAD

**Issue by CnSimonChan** [msgpack/msgpack-cli#152]
I'm using MsgPack.CLI 0.6.8 targeting .NET Framework 4.6.1

I have a class contains a private `List<int> Items` property, deserialize it throws

````
An exception of type 'System.MethodAccessException' occurred in mscorlib.dll but was not handled in user code

Additional information: Attempt by method 'MsgPack.Serialization.EmittingSerializers.Generated.MyClassSerializer43.UnpackFromCore(MsgPack.Unpacker)' to access method 'MyClass.get_Items()' failed.
````

I tried to generate serializer assembly then decompile it, I found MsgPack generated deserializing code like this:

```` C#
MyClass UnpackFromCore(Unpacker unpacker)
{
    // ....
    if (context.Items != null) // context.get_Items()
    {
        foreach (var item in unpackedList)
        {
            ((List<int>this._functionMyClass_get_Items.Invoke(context, null)).Add(item);
        }
    }
    else
    {
        this._functionMyClass_set_Items.Invoke(context, unpackedList);
    }
}
````

I think the problem is caused by the first access to `Items` directly called a `NonPublic` method `MyClass.get_Items()`. (PeVerify reported `Method is not visible`)

And the way adding item seems very inefficient.

**Comment by yfakariya** [comment]
Thank you and sorry, they are bugs.

**Comment by yfakariya** [comment]
This issue was fixed in 0.7.0-alpha3, so I've closed it.
